### DISCUSSION TITLE
Feedback - Copilot was one of the greatest helpers I've had - It degraded to a point it became a burden

### DISCUSSION THREAD

**Discussion by cmp-nct** [community/community#92433] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I've been developing with copilot since day one, been using paid APIs from OpenAI and Copilot Chat for many months for hundreds of hours.
I used it in Phpstorm but mostly in VScode on C++ projects.

In the past 3-4 weeks the service degraded to such a braindead mess, I also adapted to using it a lot and that by now costs me significantly more time than I'd need to implement everything by hand.
Copilot Chat now claims to be GPT-4, it's working at such a low level.

I've to admit, the OpenAI playground GPT-4 variants also degraded heavily in the past weeks, they were extremely useful. I paid up to  hundreds of USD per month from manual "chat" queries sent to the API. Just because the returns were good and helpful, better than Copilot was.
By now GPT-4 in Playground is barely useful, it's a huge pain to get an answer from it and by the time you have it you could have done it yourself.
Copilot Chat definitely is beyond that point, it actually doesn't provide any use anymore. It's answers are so wrong, so dumb, that any moment you interact with that is one wasted moment.

I'm leaving, will look for alternatives. the free mixtral is definitely better, deepcode is also better by now.
A month ago I would have joked on that.

**Comment by cmp-nct** [comment]
I'd like to add: 
* The new "workspace" agent is especially dumb
* Providing the context for the answer leads to answers any intern would be fired on spot
* If it answers at all, you know .. as an AI I have no access to your code .. yes sure.

![image](https://github.com/community/community/assets/78893154/b0737006-feb5-411d-89fa-eeaaa5c0533c)
![image](https://github.com/community/community/assets/78893154/41a7755e-0f9d-4143-a340-07c9e18ad249)
![image](https://github.com/community/community/assets/78893154/8ffcc277-cf5f-46c0-8829-98856e636964)
![image](https://github.com/community/community/assets/78893154/b93f7602-4964-40fb-8b47-c26cef1a4e7f)
![image](https://github.com/community/community/assets/78893154/13c639a1-c88e-451e-ae71-f94b2e475eca)
![image](https://github.com/community/community/assets/78893154/01edb5e0-fdff-4f33-baaf-66c984d10534)
![image](https://github.com/community/community/assets/78893154/267731fe-b194-45aa-b365-f178902bf45b)

**Reply by cmp-nct** [reply]
We witnessed copilot from being quite crappy to being a very powerful agentic system to becoming non accessible in June 2026. 
It has been a wild ride!

**Comment by nick-goloborodko** [comment]
I too have noticed extreme degradation of Copilot over the past few weeks. Anecdotally / ovbservationally - I've acceped about 90% of code recommendations with little or no adjustments. Now, its down to about 10% with the rest being complete rubbish. Its to the point where I hope not to receive any code recommendation prompt.

One specific scenario - Copilot used to excel at generating sample data. Now stuff it generates has no relation to the actual code. Or atleast in TypeScript and C#. Sample data it tries to generate has 80% of property names that are not present in my classes. Really frustrating.

**Comment by vjimmers** [comment]
A wonderful and terrible god it is!  Only works half the time, may delete all your code, keeps trying to get you to do things for it instead of it doing things for you.  Tells you it can't see the errors log(BS), just doesn't work at all with the dam error you would know if you used it.  Jerry's kids wrote it evidently or they hired the people who did, Not that I don't like Jerry! I was in one of his telethons when I was 12 but it can't even put a filename at the top of the code window correctly.  Keeps taking about ActiveWindows when there are none. I hate it so bad I use it 14 hours a day I hate it so much!

**Comment by pete0959** [comment]
I asked it to do a simple 'refactor' whuch was  move about 250  lines of code  into a new file and call that from the  existing...i specifically said NO functionality changes. 
8 Attempts later  it cant help but keep making breaking changes... tired of it so cancelled the subbscription

**Comment by Siccmade85** [comment]
Same experience here. Been extremely disappointed of quality and accuracy of copilot's output over past month.
Ive been power user of copilot but now i am looking for alternatives until things get resolved. Hope you dev's get things on track soon ❤️

**Comment by tamir926** [comment]
Why such great achievement step backward? If cant handle request solve by stepping forward, like upgrading infrastructure. Is there not enough profit? In my case, my pro plan suddenly changed to free plan and can't change back.

**Reply by admin** [reply]
We have an official discussion post dedicated to GitHub's move to usage-based billing where you can find detailed information, updates, and join the conversation: [GitHub Copilot is moving to usage-based billing](https://github.com/orgs/community/discussions/192948).

Please head over to that thread for the latest details and to share your feedback or questions. We'll be closing this discussion to keep things organized. Thank you for understanding! 🙏
